agent-tooling/src/main/java/io/opentelemetry/auto/typed/server/ServerTypedSpan.java

```diff
@@ -0,0 +1,20 @@
+package io.opentelemetry.auto.typed.server;
+
+import io.opentelemetry.auto.typed.base.BaseTypedSpan;
+import io.opentelemetry.trace.Span;
+
+public abstract class ServerTypedSpan<T extends ServerTypedSpan, REQUEST, RESPONSE>
+    extends BaseTypedSpan<T> {
+
+  public ServerTypedSpan(Span delegate) {
+    super(delegate);
+  }
+
+  protected abstract T onRequest(REQUEST request);
```

Comment:
I would not provide this method and instead force the user to provide the request already in startSpan, so that all attributes are set on the builder and thus available to samplers (and onStart callbacks of SpanProcessors)

Reply:
This method is not exposed to the user... it's intended for implementation by the concrete class.  Sounds like you think it's not necessary though?